src/main/java/spireMapOverhaul/zones/manasurge/events/ManaCycloneEvent.java

```diff
@@ -0,0 +1,128 @@
+                                            .filter(card -> !ManaSurgeZone.hasManaSurgeModifier(card) &&
+                                                    card.cost != -2 &&
+                                                    card.type != AbstractCard.CardType.CURSE &&
+                                                    card.type != AbstractCard.CardType.STATUS)
+                                            .forEach(filteredCards::addToTop);
+
+                                    AbstractCard randomCard = filteredCards.getRandomCard(true);
+                                    if (randomCard != null) {
+                                        ManaSurgeZone.applyPermanentNegativeModifier(randomCard);
+                                        AbstractDungeon.effectList.add(new ShowCardBrieflyEffect(randomCard.makeStatEquivalentCopy()));
+                                        filteredCards.group.remove(randomCard);
+                                    }
+                                }
+                        )
+                )
+                .addOption(OPTIONS[1],(i) -> {
+                    ArrayList<String> relicIds = new ArrayList<>();
+                    relicIds.add(BottledFlame.ID);
+                    relicIds.add(BottledTornado.ID);
+                    relicIds.add(BottledLightning.ID);
+                    relicIds.add(FrozenEgg2.ID);
+                    relicIds.add(MoltenEgg2.ID);
+                    relicIds.add(ToxicEgg2.ID);
+
+                    ArrayList<String> availableRelics = new ArrayList<>();
+                    for (String relicId : relicIds) {
+                        if (!AbstractDungeon.player.hasRelic(relicId)) {
+                            availableRelics.add(relicId);
+                        }
+                    }
+
+                    String chosenRelicId;
+                    if (!availableRelics.isEmpty()) {
+                        chosenRelicId = availableRelics.get(AbstractDungeon.miscRng.random(availableRelics.size() - 1));
+                    } else {
+                        chosenRelicId = Circlet.ID;
+                    }
+
+                    AbstractRelic chosenRelic = RelicLibrary.getRelic(chosenRelicId).makeCopy();
+                    AbstractDungeon.getCurrRoom().spawnRelicAndObtain((float) (Settings.WIDTH / 2), (float) (Settings.HEIGHT / 2), chosenRelic);
+
+                    CardGroup filteredCards = new CardGroup(CardGroup.CardGroupType.UNSPECIFIED);
+                    AbstractDungeon.player.masterDeck.group.stream()
+                            .filter(card -> !ManaSurgeZone.hasManaSurgeModifier(card) &&
+                                    card.cost != -2 &&
+                                    card.type != AbstractCard.CardType.CURSE &&
+                                    card.type != AbstractCard.CardType.STATUS)
+                            .forEach(filteredCards::addToTop);
+                    AbstractCard randomCard1 = filteredCards.getRandomCard(true);
```

Comment:
Passing true here makes it use cardRng, you need to pass the correct rng here
```suggestion
                    AbstractCard randomCard1 = filteredCards.getRandomCard(AbstractDungeon.miscRng);
```